### ISSUE TITLE
Handling of reserved identifiers and operators in the identifier parser

### ISSUE THREAD

**Issue by sjakobi** [haskell/haddock#817]
I've noticed that the identifier parser will accept e.g. `'case'` or `'..'` as identifiers although both are reserved and can't be hyperlinked. This isn't much of a problem as the renamer will mark them as `DocMonospaced` instead of `DocIdentifier` but maybe there are arguments for correcting this nonetheless?

Relatedly, maybe haddock should emit a warning when it can't resolve an identifier and applies `DocMonospaced`?

**Comment by alexbiehl** [comment]
Good call. I have prepared haddock-api to emit renamer warnings already. Feel free to make a PR to output actual warnings.

**Comment by sjakobi** [comment]
Strangely, when I tried to use my patch for #819 to produce warnings for `'case'` and `'..'`, I didn't get any.

A bit of print-debugging revealed that e.g.  `'case'` was handled as a `DocString "'case'"`. I don't understand why haddock-library's ParserSpec then parses them as `DocIdentifier`s. Any ideas?

**Comment by alexbiehl** [comment]
Maybe a difference between parseString and parseParas? Unlikely but an association I had.

**Comment by sjakobi** [comment]
> Maybe a difference between parseString and parseParas? Unlikely but an association I had.

No, `parseString` and `parseParas` both parse reserved identifiers as `DocIdentifier`.

**Comment by sjakobi** [comment]
Thanks @harpocrates. I hadn't noticed that `haddock-api` also has functions called `parseParas` and `parseString`.

I'm wondering though if users of `haddock-library` shouldn't be able to expect the parser to handle the question of valid or invalid identifiers.
